packages/engine/src/common/functions/logger.ts

```diff
@@ -48,6 +48,12 @@ const engineCache = new NodeCache()
 // Schedule the data push at a certain time (e.g., every hour)
 schedule.scheduleJob('*/15 * * * * *', pushToEngine)
 
+// Client-side early close/refres send logs
+if (typeof window !== 'undefined') {
+  console.log('Window unload event triggered. Sending logs...')
```

Comment:
This console log will fire when this file is loaded on the client. I think you want to add this console log into the event Listner function. So maybe wrap pushToEngine in a lambda/anonymousfunc and add the console log there.

Reply:
Oh ok thank you, I didn't notice, I assumed the log appearing at the top was simply a side effect of how logs are displayed.

Reply:
You're right that the console fires when the if statement is met, I have specified how to test the function in the last commit message and this is what I have done: 

- Tried modifying placement of log, still does not work
- Log does not show in vscode console due to its client sided nature
- I have removed it all together, 

If you want a log you : 

- Can send a post request to server and fire a log function in response
- Create a sort of notification on unload asking if the user wants to leave ?